src/main/java/org/example/learningsystem/multitenancy/db/service/TenantManagementService.java

```diff
@@ -0,0 +1,58 @@
+package org.example.learningsystem.multitenancy.db.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.example.learningsystem.multitenancy.context.TenantInfo;
+import org.example.learningsystem.multitenancy.db.schema.TenantSchemaService;
+import org.example.learningsystem.multitenancy.liquibase.TenantLiquibaseService;
+import org.springframework.stereotype.Service;
+
+/**
+ * Service for coordinating the onboarding and offboarding processes for tenants.
+ * <p>
+ * Manages the complete tenant onboarding and offboarding, including schema creation and deletion,
+ * data source configuration, and executing database migrations.
+ */
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class TenantManagementService {
+
+    private final TenantDataSourceService tenantDataSourceService;
+    private final TenantLiquibaseService tenantLiquibaseService;
+    private final TenantSchemaService tenantSchemaService;
+
+    /**
+     * Performs onboarding for the specified tenant.
+     *
+     * @param tenantId  the tenant identifier
+     * @param subdomain the tenant's subdomain
+     */
+    public void create(String tenantId, String subdomain) {
```

Comment:
Right now purpose of this method is not obvious (we need to rename class) and then rename method to `createSchema`